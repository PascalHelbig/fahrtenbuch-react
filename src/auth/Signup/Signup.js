import React, { Component, PropTypes } from 'react';
import Formsy from 'formsy-react';
import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
import Header from '../../Header/Header';
import EmailTextField from '../../textField/EmailTextField';
import PasswordTextField from '../../textField/PasswordTextField';


class Signup extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      values: {},
      isValid: false,
    };

    this.handleChange = this.handleChange.bind(this);
    this.signup = this.signup.bind(this);
  }

  handleChange(values) {
    this.setState({ values });
  }

  signup() {
    const { values: { email, password } } = this.state;
    this.props.onSignup(email, password);
  }

  render() {
    return (
      <div>
        <Header>Sign up</Header>
        <Formsy.Form
          onChange={this.handleChange}
          onValid={() => this.setState({ isValid: true })}
          onInvalid={() => this.setState({ isValid: false })}
        >
          <EmailTextField />
          <PasswordTextField />
        </Formsy.Form>
        <RaisedButton
          label="Registrieren"
          onTouchTap={this.signup}
          disabled={!this.state.isValid}
          primary
        />
      </div>
    );
  }
}

Signup.propTypes = {
  onSignup: PropTypes.func.isRequired,
};

export default Signup;
