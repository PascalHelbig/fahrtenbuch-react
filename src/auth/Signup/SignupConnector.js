import React, { PropTypes } from 'react';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import Signup from './Signup';

export const SignupConnectorDumb = ({ signup }) => (
  <Signup onSignup={(email, password) => signup({ variables: { email, password } })} />
);

SignupConnectorDumb.propTypes = {
  signup: PropTypes.func.isRequired,
};

const signupMutation = gql`mutation signup($email: Email!, $password: Password!) {
    signup( email: $email, password: $password ) {
        token
    }
}`;

export default graphql(signupMutation, { name: 'signup' })(SignupConnectorDumb);
