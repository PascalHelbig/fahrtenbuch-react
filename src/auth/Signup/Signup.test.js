import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
import Formsy from 'formsy-react';
import Signup from './Signup';

it('renders correctly', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('runs signup on button touchTap', () => {
  const onSignup = jest.fn();
  const wrapper = shallow(<Signup onSignup={onSignup} />);
  wrapper.find(RaisedButton).simulate('touchTap');
  expect(onSignup).toHaveBeenCalled();
});

it('enable button on valid form', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  wrapper.setState({ isValid: true });

  const signupButton = wrapper.find(RaisedButton);
  expect(signupButton.prop('disabled')).toBeFalsy();
});

it('disable button on invalid from', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  wrapper.setState({ isValid: false });

  const signupButton = wrapper.find(RaisedButton);
  expect(signupButton.prop('disabled')).toBeTruthy();
});

it('saves changes in state', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  wrapper.instance().handleChange({ email: 'test@test.de' });

  expect(wrapper.state('values')).toEqual({ email: 'test@test.de' });
});

it('set isValid state to true, when form calls onValid()', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  const form = wrapper.find(Formsy.Form);

  wrapper.setState({ isValid: false });
  form.simulate('valid');

  expect(wrapper.state('isValid')).toBeTruthy();
});

it('set isValid state to false, when form calls onInvalid()', () => {
  const wrapper = shallow(<Signup onSignup={() => {}} />);
  const form = wrapper.find(Formsy.Form);

  wrapper.setState({ isValid: true });
  form.simulate('invalid');

  expect(wrapper.state('isValid')).toBeFalsy();
});