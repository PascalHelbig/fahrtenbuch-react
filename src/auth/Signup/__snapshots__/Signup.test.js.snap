exports[`test renders correctly 1`] = `
<div>
  <Header>
    Sign up
  </Header>
  <Formsy
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onSubmit={[Function]}
    onSuccess={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    validationErrors={null}>
    <Component />
    <Component />
  </Formsy>
  <RaisedButton
    disabled={true}
    fullWidth={false}
    label="Registrieren"
    labelPosition="after"
    onTouchTap={[Function]}
    primary={true}
    secondary={false} />
</div>
`;
